agent/config_endpoint_test.go

```diff
@@ -95,6 +108,37 @@ func TestConfig_Get(t *testing.T) {
 		_, err := a.srv.Config(resp, req)
 		require.Error(t, errors.New("Must provide either a kind or both kind and name"), err)
 	})
+	t.Run("get the single mesh config", func(t *testing.T) {
+		req, _ := http.NewRequest("GET", "/v1/config/mesh/mesh", nil)
```

Comment:
Querying for the entry still required me to specify the "name" in the URL because we haven't special cased this type in the memdb indexer yet. An example of that can be seen in #10067. This would allow someone to query for `/v1/config/mesh/`